### PR TITLE
Add DirichletMultinomial distribution

### PR DESCRIPTION
**Pull request by fehiepsi**
This distribution is required for some conjugacy tests of `collapse` messenger.

Also addresses #703.

### CODE REVIEW

File: numpyro/distributions/conjugate.py [hunk 1]
```diff
@@ -79,6 +81,69 @@ def enumerate_support(self, expand=True):
...
+    classes (``probs`` for the :class:`~numpyro.distributions.Multinomial` distribution)
+    is unknown and randomly drawn from a :class:`~numpyro.distributions.Dirichlet`
+    distribution prior to a certain number of Categorical trials given by
+    ``total_count``.
+
+    :param numpy.ndarray concentration: concentration parameter (alpha) for the
+        Dirichlet distribution.
+    :param numpy.ndarray total_count: number of Categorical trials.
+    :param bool is_sparse: Whether to assume value is mostly zero when computing
+        :meth:`log_prob`, which can speed up computation when data is sparse.
```
Comment (fritzo) [inline numpyro/distributions/conjugate.py:96]: remove `is_sparse` docs

**Review by fritzo — COMMENTED** [review]
LGTM except for the minor docs fix.

File: numpyro/distributions/conjugate.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/distributions/conjugate.py]: good catch, thanks, Fritz!

File: numpyro/distributions/conjugate.py [hunk 2]
```diff
@@ -79,6 +81,69 @@ def enumerate_support(self, expand=True):
...
+    def variance(self):
+        n = jnp.expand_dims(self.total_count, -1)
+        alpha = self.concentration
+        alpha_sum = self.concentration.sum(-1, keepdims=True)
+        alpha_ratio = alpha / alpha_sum
+        return n * alpha_ratio * (1 - alpha_ratio) * (n + alpha_sum) / (1 + alpha_sum)
+
+    @property
+    def support(self):
+        return constraints.multinomial(self.total_count)
```
Comment (fritzo) [inline numpyro/distributions/conjugate.py:142]: Nice, we should add a `multionomial` constraint to torch.

**Review by fritzo — APPROVED** [review]

File: numpyro/distributions/conjugate.py (hunk 2 above)
Comment (fehiepsi) [reply numpyro/distributions/conjugate.py:142]: cc @neerajprad who invented this constraint :D

**Pull request merged**